x-pack/plugin/search-business-rules/src/main/java/org/apache/lucene/search/CappedScoreQuery.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.apache.lucene.search;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.util.Bits;
+
+/**
+ * A query that wraps another query and ensures scores do not exceed a maximum value
+ */
+public final class CappedScoreQuery extends Query {
+    private final Query query;
+    private final float maxScore;
+
+    /** Caps scores from the passed in Query to the supplied maxScore parameter */
+    public CappedScoreQuery(Query query, float maxScore) {
+        this.query = Objects.requireNonNull(query, "Query must not be null");
+        if (maxScore <= 0) {
+            throw new IllegalArgumentException(this.getClass().getName() + " maxScore must be >0, " + maxScore + " supplied.");
+        }
+        this.maxScore = maxScore;
+    }
+
+    /** Returns the encapsulated query. */
+    public Query getQuery() {
+        return query;
+    }
+
+    @Override
+    public Query rewrite(IndexReader reader) throws IOException {
+        Query rewritten = query.rewrite(reader);
+
+        if (rewritten != query) {
+            return new CappedScoreQuery(rewritten, maxScore);
+        }
+
+        if (rewritten.getClass() == CappedScoreQuery.class) {
+            return rewritten;
+        }
+
+        if (rewritten.getClass() == BoostQuery.class) {
+            return new CappedScoreQuery(((BoostQuery) rewritten).getQuery(), maxScore);
+        }
```

Comment:
I don't think removing the BoostQuery is correct?

Reply:
Another copy-and-paste. This time from ConstantScoreQuery which seemed the closest function to CappedScoreQuery